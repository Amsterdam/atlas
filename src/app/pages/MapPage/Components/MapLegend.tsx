import { MapPanelContent } from '@datapunt/arm-core'
import React from 'react'

import styled from 'styled-components'
import MapPanel from '../../../../map/containers/panel/MapPanel'
import getState from '../../../../shared/services/redux/get-state'
import MapContext from '../MapContext'

const StyledMapPanelContent = styled(MapPanelContent)`
  overflow: hidden; // This can be removed if the new design for the legend is added

  .map-panel {
    max-height: 100%;
    height: 100%;
    max-width: 100%;
    width: 100%;
    box-shadow: none;
    margin: 0;
    bottom: 0;
    left: 0;

    & > .scroll-wrapper {
      max-height: 100%;
      height: 100%;
    }

    & > .map-panel__heading {
      display: none;
    }
  }
`

const MapLegend: React.FC = ({ ...otherProps }) => {
  const {
    panelLayers,
    activeMapLayers,
    zoomLevel,
    setActiveMapLayers,
    setVisibleMapLayer,
  } = React.useContext(MapContext)

  // Get the user from the Redux state for now
  const { user } = getState()

  const onLayerToggle = React.useCallback(
    ({
      id,
      legendItems,
    }: {
      // Add auto type generation
      id: string
      legendItems: Array<{ id: string; notSelectable: boolean }>
    }) => {
      if (setActiveMapLayers)
        setActiveMapLayers(
          legendItems?.some(({ notSelectable }) => !notSelectable) && legendItems.length > 0
            ? legendItems.map(({ id: legendItemId }) => ({
                id: legendItemId,
                isVisible: true,
              }))
            : [{ id, isVisible: true }],
        )
    },
    [setActiveMapLayers],
  )

  return (
    <StyledMapPanelContent title="Legenda" {...otherProps}>
      <MapPanel
        panelLayers={panelLayers}
        overlays={activeMapLayers}
        onLayerToggle={onLayerToggle}
        onLayerVisibilityToggle={setVisibleMapLayer}
        isMapPanelVisible
        zoomLevel={zoomLevel}
        user={user}
      />
    </StyledMapPanelContent>
  )
}

export default MapLegend
