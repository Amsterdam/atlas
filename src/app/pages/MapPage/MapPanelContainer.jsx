import React from 'react'
import MapContext from './MapContext'

// This is the "old" MapPanel that will be replaced
import MapPanel from '../../../map/containers/panel/MapPanel'
import getState from '../../../shared/services/redux/get-state'

const MapPanelComponent = () => {
  const {
    baseLayers,
    panelLayers,
    activeBaseLayer,
    activeMapLayers,
    isMapPanelVisible,
    zoomLevel,
    setActiveBaseLayer,
    setActiveMapLayers,
    setVisibleMapLayers,
    setMapPanelVisible,
    getBaseLayers,
    getPanelLayers,
  } = React.useContext(MapContext)

  // Get the user from the Redux state for now
  const { user } = getState()

  React.useEffect(() => {
    getBaseLayers()
    getPanelLayers()
  }, [])

  function onLayerToggle(payload) {
    setActiveMapLayers(
      payload.legendItems?.some(({ notSelectable }) => !notSelectable) &&
        payload.legendItems.length > 0
        ? payload.legendItems.map(({ id: legendItemId }) => ({ id: legendItemId, isVisible: true }))
        : [{ id: payload.id, isVisible: true }],
    )
  }

  return (
    <MapPanel
      mapBaseLayers={baseLayers}
      panelLayers={panelLayers}
      activeBaseLayer={activeBaseLayer}
      overlays={activeMapLayers}
      onBaseLayerToggle={setActiveBaseLayer}
      onLayerToggle={onLayerToggle}
      onLayerVisibilityToggle={setVisibleMapLayers}
      onMapPanelToggle={setMapPanelVisible}
      isMapPanelVisible={isMapPanelVisible}
      zoomLevel={zoomLevel}
      user={user}
    />
  )
}

export default MapPanelComponent
