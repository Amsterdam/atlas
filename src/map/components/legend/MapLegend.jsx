import React from 'react';
import PropTypes from 'prop-types';

import { Checkbox } from '../../../shared/components/checkbox';

const MapLegend = ({ activeMapLayers }) => (
  <ul className="map-legend">
    {activeMapLayers.map(mapLayer => (
      <li key={mapLayer.id}>
        <div className="map-legend__category">
          <Checkbox
            checked="true"
            name={mapLayer.id}
          />
          <span>{mapLayer.title}</span>
        </div>
        <ul className="map-legend__items">
          {/* TODO: Remove `.filter()` as soon data is complete */}
          {mapLayer.legendItems.filter(legendItem => !!legendItem.title).map(legendItem => (
            <li
              className="map-legend__item"
              key={legendItem.id}
            >
              <Checkbox
                checked="true"
                name={legendItem.id}
              />
              <div className="map-legend__image">
                <img
                  alt=""
                  src={legendItem.imageUrl}
                />
              </div>
              <span>{legendItem.title}</span>
            </li>
          ))}
        </ul>
      </li>
    ))}
  </ul>
);

MapLegend.propTypes = {
  activeMapLayers: PropTypes.array, // eslint-disable-line
};

export default MapLegend;
