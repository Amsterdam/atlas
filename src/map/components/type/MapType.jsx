import React from 'react';
import PropTypes from 'prop-types';

import { SelectButton } from '../../../shared/components/selectbutton';

import AerialIcon from '../../../../public/images/icon-aerial.svg';
import TopographyIcon from '../../../../public/images/icon-topography.svg';

function hasLayer(activeLayer, layers) {
  return layers && layers.length > 0 && layers.some(l => l.value === activeLayer);
}

const MapType = ({ layers, activeLayer, onBaseLayerToggle }) => (
  <section className="map-panel__map-type">
    <div className="map-panel__map-type-title">Achtergrond</div>

    <SelectButton
      className="map-panel__map-type-select map-panel__map-type-select--topography"
      name="topography"
      icon={TopographyIcon}
      options={layers.topography}
      value={activeLayer}
      disabled={!hasLayer(activeLayer, layers.topography)}
      handleChange={onBaseLayerToggle}
    />

    <SelectButton
      className="map-panel__map-type-select map-panel__map-type-select--aerial"
      name="aerial"
      icon={AerialIcon}
      options={layers.aerial}
      value={activeLayer}
      disabled={!hasLayer(activeLayer, layers.aerial)}
      handleChange={onBaseLayerToggle}
    />
  </section>
);

MapType.propTypes = {
  layers: PropTypes.object, // eslint-disable-line
  activeLayer: PropTypes.string, // eslint-disable-line
  onBaseLayerToggle: PropTypes.func // eslint-disable-line
};

export default MapType;
