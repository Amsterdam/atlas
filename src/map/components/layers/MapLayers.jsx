import React from 'react';
import PropTypes from 'prop-types';

import MapLegend from '../legend/MapLegend';
import Notification from '../../../shared/components/notification/Notification';
import AddIcon from '../../../../public/images/icon-plus.svg';
import RemoveIcon from '../../../../public/images/icon-cross.svg';

import './_map-layers.scss';

const showLayer = (layer, user) => (
  !layer.authScope || (user.authenticated && user.scopes.includes(layer.authScope))
);

const showCategory = (layers, user) => (
  layers.filter((layer) => layer.category && showLayer(layer, user))
);

const MapLayers = (props) => (
  <div className="map-layers">
    <h3 className="u-sr-only">Beschikbare kaartlagen</h3>
    <ul>
      {[...new Set(showCategory(props.layers, props.user).map((layer) => layer.category))]
        .map((category) => (
          <li className="map-layers__category" key={category}>
            <h4 className="map-layers__category-text">{category}</h4>
            <ul>
              {props.layers
                .filter((layer) => layer.category === category && showLayer(layer, props.user))
                .map((layer) => (
                  <li
                    className={`
                      map-layers__title
                      map-layers__title--${props.activeMapLayers.some((mapLayer) => layer.title === mapLayer.title) ? 'active' : 'inactive'}
                    `}
                    key={layer.title}
                  >
                    <button onClick={() => {
                      MapLegend.mapLayersLegendItemsToIds(layer).forEach((mapLayerId) =>
                        props.onLayerToggle(mapLayerId));
                    }}
                    >
                      <span className="map-layers__toggle-title">
                        {layer.title}
                      </span>
                      <span className="map-layers__toggle map-layers__toggle--remove">
                        <RemoveIcon />
                      </span>
                      <span className="map-layers__toggle map-layers__toggle--add">
                        <AddIcon />
                      </span>
                    </button>
                  </li>
              ))}
            </ul>
          </li>
      ))}
    </ul>
    {!props.user.authenticated && (
      <Notification>
        <a
          href="#?mpb=topografie&pgn=content-overzicht&pgi=item1&pgt=inloggen"
        >Meer kaartlagen</a> na inloggen.
      </Notification>
    )}
  </div>
);

MapLayers.propTypes = {
  activeMapLayers: PropTypes.array, // eslint-disable-line
  layers: PropTypes.array, // eslint-disable-line
  onLayerToggle: PropTypes.func.isRequired, // eslint-disable-line react/no-unused-prop-types
  user: PropTypes.object // eslint-disable-line
};

export default MapLayers;
