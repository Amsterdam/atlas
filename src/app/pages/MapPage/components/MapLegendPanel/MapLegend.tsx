import { themeColor } from '@amsterdam/asc-ui'
import { useMatomo } from '@datapunt/matomo-tracker-react'
import type { FunctionComponent } from 'react'
import { useEffect, useMemo, useState } from 'react'
import styled from 'styled-components'
import { useIsEmbedded } from '../../../../contexts/ui'
import MapCollectionButton from './MapCollectionButton'
import MapLayerWithLegendItem from './MapLayerWithLegendItem'
import MapLayerButton from './MapLayerButton'
import type { ExtendedMapGroup } from '../../legacy/services'

const MapCollectionList = styled.ul`
  background-color: ${themeColor('tint', 'level1')};
  border-top: none;
  list-style-type: none;
  padding: 0;
  user-select: none;
`

export interface MapLegendProps {
  activeMapLayers: ExtendedMapGroup[]
  title: string
  onAddLayers: (layers: string[]) => void
  onRemoveLayers: (layers: string[]) => void
}

// Todo: legacy component, should be refactored / simplified
const MapLegend: FunctionComponent<MapLegendProps> = ({
  activeMapLayers,
  title,
  onRemoveLayers,
  onAddLayers,
}) => {
  const { trackEvent } = useMatomo()
  const isEmbedView = useIsEmbedded()

  function trackLayerEnabled(mapLayer: ExtendedMapGroup) {
    // Sanitize the collection title to use it as action
    const action = mapLayer.title.toLowerCase().replace(/[: ][ ]*/g, '_')

    if (!mapLayer.legendItems || mapLayer.legendItems.length === 0) {
      trackEvent({
        category: 'kaartlaag',
        action,
        name: mapLayer.title,
      })
    } else {
      mapLayer.legendItems.forEach(({ title: legendItemTitle }) =>
        trackEvent({
          category: 'kaartlaag',
          action,
          name: legendItemTitle,
        }),
      )
    }
  }

  // Todo: This need to be refactored on redux level, so MapLayers and LegendItems will get a isVisible option there.
  const mapLayers = activeMapLayers

  const [collectionOpen, setCollectionOpen] = useState(isEmbedView ?? false)

  const allVisible = mapLayers.every(({ isVisible }) => isVisible)
  const someVisible = mapLayers.some(({ isVisible }) => isVisible)
  const collectionIndeterminate =
    mapLayers.some(
      ({ isVisible, legendItems }) =>
        !isVisible ||
        legendItems?.some(({ isVisible: legendVisibility }: any) => !legendVisibility),
    ) && someVisible

  const handleLayerToggle = (checked: boolean, mapLayer: ExtendedMapGroup) => {
    // Only track the event when the mapGroup gets checked
    if (checked) {
      trackLayerEnabled(mapLayer)
    }
  }

  // Check if some of the maplayers are opened on first render
  const hasOpenMapLayers = useMemo(
    () =>
      mapLayers.some(
        ({ isVisible, legendItems }) =>
          isVisible ||
          (isVisible &&
            legendItems?.some(({ isVisible: legendItemIsVisible }: any) => legendItemIsVisible)), // legenditems could be visible in another MapLegend
      ),
    [mapLayers],
  )

  // Open the maplegend if some of the maplayers are opened
  useEffect(() => {
    if (hasOpenMapLayers) setCollectionOpen(true)
  }, [hasOpenMapLayers])

  const addOrRemoveLayer = (
    checked: boolean,
    layers: ExtendedMapGroup[],
    onlyInvisible = false,
  ) => {
    const layersToFilter = onlyInvisible ? layers.filter(({ isVisible }) => !isVisible) : layers
    const filteredMapLayers = layersToFilter
      .map((mapLayer) => {
        if (mapLayer?.legendItems?.length && mapLayer?.legendItems?.every(({ id }) => id)) {
          return mapLayer?.legendItems?.map((legendItem) => legendItem.id ?? '')
        }
        return mapLayer.id
      })
      .flat()
      .filter((id) => id)

    if (checked) {
      if (onAddLayers) {
        onAddLayers(filteredMapLayers)
      }
    } else if (onRemoveLayers) {
      onRemoveLayers(filteredMapLayers)
    }
  }

  return (
    <>
      {(!isEmbedView || (isEmbedView && mapLayers.some(({ isEmbedded }) => isEmbedded))) && ( // Also display the collection title when maplayer is embedded
        <MapCollectionButton
          collectionIndeterminate={collectionIndeterminate}
          isOpen={collectionOpen}
          onClick={() => setCollectionOpen(!collectionOpen)}
          allVisible={allVisible}
          title={title}
        />
      )}
      {collectionOpen && (
        <MapCollectionList>
          {mapLayers &&
            mapLayers
              .filter(({ isEmbedded }) => (isEmbedView ? isEmbedded : true)) // Only include the embedded layers when in certain views
              .map((mapGroup) => {
                const layerIsChecked = mapGroup.isVisible
                const layerIsIndeterminate =
                  mapGroup.legendItems?.some(({ isVisible }: any) => !isVisible) && layerIsChecked

                const hasLegendItems = mapGroup?.legendItems && mapGroup.legendItems.length > 0

                if (!hasLegendItems) {
                  return (
                    <MapLayerWithLegendItem
                      key={mapGroup.id}
                      onAddLayers={onAddLayers}
                      onRemoveLayers={onRemoveLayers}
                      legendItem={mapGroup}
                    />
                  )
                }

                return (
                  <MapLayerButton
                    key={mapGroup.id}
                    mapGroup={mapGroup}
                    layerIsChecked={layerIsChecked}
                    layerIsIndeterminate={layerIsIndeterminate}
                    addOrRemoveLayer={addOrRemoveLayer}
                    isEmbedView={isEmbedView}
                    handleLayerToggle={handleLayerToggle}
                    onAddLayers={onAddLayers}
                    onRemoveLayers={onRemoveLayers}
                  />
                )
              })}
        </MapCollectionList>
      )}
    </>
  )
}

export default MapLegend
