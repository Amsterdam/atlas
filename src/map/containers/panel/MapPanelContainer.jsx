import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { getBaseLayers } from '../../ducks/layers/base-layers';
import { setBaseLayer } from '../../ducks/base-layer/base-layer';
import { getMapLayers, selectActiveMapLayers } from '../../ducks/layers/map-layers';
import { toggleMapOverlay, toggleMapOverlayVisibility } from '../../ducks/overlays/overlays';
import MapLayers from '../../components/layers/MapLayers';
import MapLegend from '../../components/legend/MapLegend';
import MapType from '../../components/type/MapType';
import MapLayersIcon from '../../../../public/images/icon-map-layers.svg';

const mapStateToProps = state => ({
  activeMapLayers: selectActiveMapLayers(state),
  atlas: state.atlas,
  layerSelection: state.layerSelection,
  baseLayers: state.baseLayers,
  baseLayer: state.map.baseLayer,
  mapLayers: state.mapLayers,
  overlays: state.map.overlays,
  zoomLevel: state.map.zoom
});

// TODO: Add method that checks whether layer is active and toggles accordingly
const mapDispatchToProps = dispatch => bindActionCreators({
  onLayerToggle: toggleMapOverlay,
  onBaseLayerToggle: setBaseLayer,
  onLayersToggle: toggleMapOverlay,
  onLayerVisibilityToggle: toggleMapOverlayVisibility
}, dispatch);

class MapPanelContainer extends React.Component {
  componentDidMount() {
    this.context.store.dispatch(getBaseLayers());
    this.context.store.dispatch(getMapLayers());
  }

  render() {
    return this.props.layerSelection.isEnabled && (
      <section className="map-panel">
        <div className="map-panel__heading">
          <MapLayersIcon className="map-panel__heading-icon" />
          <h1 className="map-panel__heading-title">Kaartlagen</h1>
        </div>
        <div className="scroll-wrapper">
          {this.props.activeMapLayers.length > 0 && (
            <MapLegend
              activeMapLayers={this.props.activeMapLayers}
              onLayerToggle={this.props.onLayerToggle}
              onLayerVisibilityToggle={this.props.onLayerVisibilityToggle}
              overlays={this.props.overlays}
              zoomLevel={this.props.zoomLevel}
            />
          )}
          <MapType
            layers={this.props.baseLayers}
            activeLayer={this.props.baseLayer}
            onBaseLayerToggle={this.props.onBaseLayerToggle}
          />
          <MapLayers
            activeMapLayers={this.props.activeMapLayers}
            layers={this.props.mapLayers}
            onLayerToggle={this.props.onLayerToggle}
          />
        </div>
      </section>
    );
  }
}

MapPanelContainer.contextTypes = {
  store: PropTypes.object.isRequired
};

MapPanelContainer.defaultProps = {
  activeMapLayers: [],
  atlas: {},
  layerSelection: {},
  map: {},
  baseLayers: {},
  mapLayers: [],
  zoomLevel: 0
};

MapPanelContainer.propTypes = {
  activeMapLayers: PropTypes.array, // eslint-disable-line
  atlas: PropTypes.object, // eslint-disable-line
  layerSelection: PropTypes.object, // eslint-disable-line
  map: PropTypes.object, // eslint-disable-line
  baseLayers: PropTypes.object, // eslint-disable-line
  mapLayers: PropTypes.array, // eslint-disable-line
  onLayerToggle: PropTypes.func, // eslint-disable-line
  onBaseLayerToggle: PropTypes.func, // eslint-disable-line
  onLayersToggle: PropTypes.func, // eslint-disable-line
  onLayerVisibilityToggle: PropTypes.func, // eslint-disable-line
  overlays: PropTypes.array, // eslint-disable-line
  zoomLevel: PropTypes.number
};

export default connect(mapStateToProps, mapDispatchToProps)(MapPanelContainer);
