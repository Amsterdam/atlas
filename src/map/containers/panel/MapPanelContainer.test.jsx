import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import MapPanelContainer from './MapPanelContainer';
import MapLayers from '../../components/layers/MapLayers';
import MapLegend from '../../components/legend/MapLegend';
import MapType from '../../components/type/MapType';

describe('MapPanelContainer', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = configureMockStore()({
      map: {
        baseLayer: '',
        overlays: [{
          isVisible: true
        }]
      },
      mapLayers: {
        layers: {
          items: []
        },
        baseLayers: {
          items: []
        },
        panelLayers: {
          items: []
        }
      },
      overlays: [{}],
      ui: { isMapPanelHandleVisible: true }
    });
    wrapper = shallow(<MapPanelContainer />, { context: { store } }).dive();
  });

  xit('should render MapType and MapLayers', () => {
    expect(wrapper.find(MapType).length).toBe(1);
    expect(wrapper.find(MapLayers).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  xit('should render MapLegend if store contains active map layers', () => {
    expect(wrapper.find(MapLegend).length).toBe(0);
    wrapper.setProps({ activeMapLayers: [{}] });
    expect(wrapper.find(MapLegend).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  xit('should scroll the map-legend is map panel is visible and have more overlays', () => {
    const scrollIntoViewMock = jest.fn();
    document.querySelector = jest.fn().mockReturnValue({
      scrollIntoView: scrollIntoViewMock
    });
    wrapper.setProps({
      overlays: [{}, {}],
      isMapPanelVisible: true
    });
    expect(scrollIntoViewMock).toHaveBeenCalled();
  });
});
