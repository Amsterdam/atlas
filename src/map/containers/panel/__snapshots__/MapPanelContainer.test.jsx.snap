// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapPanelContainer should render MapLegend if store contains active map layers 1`] = `
<section
  className="
        map-panel
        map-panel--collapsed
        map-panel--has--active-layers
      "
>
  <div
    className="map-panel__heading"
  >
    <test-file-stub
      className="map-panel__heading-icon"
    />
    <h1
      className="map-panel__heading-title"
    >
      Kaartlagen
    </h1>
    <button
      className="map-panel__toggle"
      onClick={[Function]}
    >
      <test-file-stub
        className="map-panel__toggle-icon map-panel__toggle-icon--expanded"
      />
      <test-file-stub
        className="map-panel__toggle-icon map-panel__toggle-icon--collapsed"
      />
    </button>
  </div>
  <div
    className="scroll-wrapper"
  >
    <MapLegend
      activeMapLayers={
        Array [
          Object {},
        ]
      }
      onLayerToggle={[Function]}
      onLayerVisibilityToggle={[Function]}
      overlays={Array []}
      zoomLevel={0}
    />
    <MapType
      activeBaseLayer=""
      baseLayers={Object {}}
      onBaseLayerToggle={[Function]}
    />
    <MapLayers
      activeMapLayers={
        Array [
          Object {},
        ]
      }
      layers={
        Array [
          Object {
            "fiets": "hoi",
          },
        ]
      }
      onLayerToggle={[Function]}
      user={Object {}}
    />
  </div>
</section>
`;

exports[`MapPanelContainer should render MapType and MapLayers 1`] = `
<section
  className="
        map-panel
        map-panel--collapsed
        map-panel--has-no-active-layers
      "
>
  <div
    className="map-panel__heading"
  >
    <test-file-stub
      className="map-panel__heading-icon"
    />
    <h1
      className="map-panel__heading-title"
    >
      Kaartlagen
    </h1>
    <button
      className="map-panel__toggle"
      onClick={[Function]}
    >
      <test-file-stub
        className="map-panel__toggle-icon map-panel__toggle-icon--expanded"
      />
      <test-file-stub
        className="map-panel__toggle-icon map-panel__toggle-icon--collapsed"
      />
    </button>
  </div>
  <div
    className="scroll-wrapper"
  >
    <MapType
      activeBaseLayer=""
      baseLayers={Object {}}
      onBaseLayerToggle={[Function]}
    />
    <MapLayers
      activeMapLayers={Array []}
      layers={
        Array [
          Object {
            "fiets": "hoi",
          },
        ]
      }
      onLayerToggle={[Function]}
      user={Object {}}
    />
  </div>
</section>
`;
